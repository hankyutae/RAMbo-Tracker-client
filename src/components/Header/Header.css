a{
  text-decoration: none;
  color: rgb(0, 0, 0);
  display:block;
  text-align: right;
}


#main-title{
  text-align:left;
  border:none;
  font-size: 40px;
  font-family: 'Dancing Script', cursive;
  color: black;
  position: relative;
  top: -5px;
}

.user-container{
  padding-right: 2%;
  width: 100%;
  margin: 0 auto;
  font-weight: bold;
  
}
.Header {
  display: flex;
  width: 40%;
  margin: 0 0 0 10px;
}

.nav-container{
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  padding: 0px;
  width: 80vw;
  height: 110px;
  margin-left: 10px;
  background-color: rgb(11, 76, 114);
  color: white;
  border-radius: 0px 0px 0px 25px;
}

header {
  padding: 0px;
  font-size: 20px;
  display: flex;
  align-items: center;
  width: 100%;
  margin-top: 0px;
}

#user-name{
  font-size: 27px;
  color: #3369ff;
}

.user-container{
  text-align: right;
}

.LoginLogout{
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  padding: 10px;
  font-size: 20px;
}

#logoutlink{
  color: white;
  margin: 0 auto;
}

#logoutlink:hover{
  color: white;
}

#loginlink, #dashlink{
  padding: 28px 10px;
  color: white;
}

#aboutlink{
  margin: 0 auto;
}

#signuplink{
  padding: 0px 10px;
  color: white;
  margin: 0 auto;
}

.header-photo{
  width: 90px;
  height: 100px;
  margin: -10px 0px 0px -10px;
}

@media screen and (max-width: 420px) and (min-width: 310px){

  .header-photo{
    width: 60px;
    height: 70px;
    margin: -10px 0px 0px -5px;
  }
  
  header {
    padding: 0px;
    margin: 10px 0 0 0;
    font-size: 20px;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    width: 100%;
  }

  .LoginLogout{
    padding: 10px;
    font-size: 20px;
  }
  
  .Header{
    display: flex;
    justify-content: center;
    justify-content: space-evenly;
    width: 100%;
  }
  
  #loginlink, #dashlink{
    padding: 0 0 0 0;
    color: white;
    margin: 0 20px 0 20px;
  }

  #signuplink{
    padding: 0px 0px;
    color: white;
    margin: 0 20px 0 20px;
  }

  #aboutlink{
    margin: 0 20px 0 20px;
  }

  #logoutlink{
    margin: 0 20px 0 20px;
  }

  #main-title{
    text-align:center;
    white-space: nowrap;
    border: none;
    font-size: 36px;
    font-family: 'Dancing Script', cursive;
    color: black;
    margin: 0 5px 0 0;
  }

  .nav-container{
    display: flex;
    justify-content: space-around;
    position: absolute;
    top: 65px;
    padding: 0px;
    width: 100vw;
    height: 50px;
    margin: 0px 0px 0px 0px;
    border-radius: 0px;

  }

  .registration-form{
    margin-top: 50px;
  }
 
} 

@media screen and (min-width: 421px) and (max-width: 1020px){

  .header-photo{
    width: 80px;
    height: 90px;
    margin: -20px 0px 0px 5px;
  }

  .LoginLogout{
    padding: 10px;
    font-size: 20px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
  }
  
  .Header{
    width: 80%;
    margin: 0 0 0 10px;
  }
  
  #loginlink, #dashlink{
    padding: 28px 10px;
    color: white;
  }

  #signuplink{
    padding: 0px 10px;
  }

  #logoutlink{
    margin: 0 auto;
  }

  #aboutlink{
    margin: 0 auto;
  }

  #main-title{
    text-align:left;
    border:none;
    font-size: 36px;
    font-family: 'Dancing Script', cursive;
    margin-left: 10px;
    position: relative;
    top: -10px;
  }

  header {
    margin: -2px -5px 0px 0px;
    padding: 0px;
    font-size: 20px;
    display: flex;
    align-items: center;
    width: 100%;
  }


  .nav-container{
    display: flex;
    align-items: center;
    padding: 0px;
    width: 100vw;
    height: 118px;
    margin: 0px -1px 0px 0px;
  }

  
 
} 

  