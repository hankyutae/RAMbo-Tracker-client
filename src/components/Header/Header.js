import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import './Header.css'

class Header extends Component {
  state={
    refresh:true
  }
  handleLogoutClick = () => {
    TokenService.clearAuthToken();this.setState({
      refresh: !this.state.refresh
    })
  }

  renderLogoutLink() {
    return (
      <div className='user-container'>
        <nav>
          <Link
            onClick={this.handleLogoutClick}
            to='/login'>
            Logout
          </Link>
        </nav>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='nav-container'>
      <nav className='LoginLogout'>
        <Link id='loginlink' to='/login'>Login</Link>
        {' '}
        <Link id='signuplink' to='/register'>Sign up</Link>
      </nav>
      </div>
    )
  }

  render() {
    return (
      <header>
        <h1 className='Header'>
          <Link id='main-title' to='/'>
            Symptom Tracker
          </Link>
        </h1>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
          <div id='box'></div>
      </header>
    );
  }
}

export default Header