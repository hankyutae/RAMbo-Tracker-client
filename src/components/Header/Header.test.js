import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Header refreshesWhenAppStateDoes={() => {}} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
