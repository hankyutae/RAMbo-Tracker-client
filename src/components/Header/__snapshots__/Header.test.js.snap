// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <nav>
    <h1
      className="Header"
    >
      <a
        href="/"
        id="main-title"
        onClick={[Function]}
      >
        Symptom Tracker
      </a>
    </h1>
    <div
      className="nav-container"
    >
      <a
        href="/login"
        id="loginlink"
        onClick={[Function]}
      >
        Login
      </a>
       
      <a
        href="/register"
        id="signuplink"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
    <div
      id="box"
    />
  </nav>
</div>
`;
